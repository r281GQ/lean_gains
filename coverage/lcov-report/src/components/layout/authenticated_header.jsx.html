<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/authenticated_header.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> authenticated_header.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Toolbar } from 'material-ui';
&nbsp;
import RouterNavigation from './router_navigation';
import Main from './main';
import Exercises from './exercises';
import User from './user';
import Macros from './macros';
import SaveIndicator from './save_indicator';
&nbsp;
&nbsp;
&nbsp;
const AuthenticatedHeader = <span class="fstat-no" title="function not covered" >({</span>
  openSideBar,
  exercises,
  isFetching,
  isLoading,
  userName,
  photo,
  todaysMacros,
  goBack,
  goForward
}) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;Toolbar style={{ backgroundColor: '#757575', color: '#EEEEEE' }}&gt;</span>
    &lt;Main openSideBar={openSideBar} /&gt;
    &lt;RouterNavigation goBack={goBack} goForward={goForward} /&gt;
    &lt;Exercises exercises={exercises} /&gt;
    &lt;Macros todaysMacros={todaysMacros} isFetching={isFetching} /&gt;
    &lt;SaveIndicator isLoading={isLoading} /&gt;
    &lt;User userName={userName} photo={photo} /&gt;
  &lt;/Toolbar&gt;;
&nbsp;
AuthenticatedHeader.propTypes = {
  openSideBar: PropTypes.func,
  exercises: PropTypes.arrayOf(PropTypes.string),
  isFetching: PropTypes.bool,
  isLoading: PropTypes.bool,
  userName: PropTypes.string,
  photo: PropTypes.string,
  todaysMacros: PropTypes.object
};
&nbsp;
export default AuthenticatedHeader;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
