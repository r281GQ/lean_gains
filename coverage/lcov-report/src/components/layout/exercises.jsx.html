<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/exercises.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> exercises.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import * as _ from 'lodash';
import {
  IconButton,
  ToolbarGroup,
  ToolbarTitle,
  MenuItem,
  IconMenu
} from 'material-ui';
import ContentFilter from 'material-ui/svg-icons/navigation/arrow-downward';
&nbsp;
const Exercises = <span class="fstat-no" title="function not covered" >({</span> exercises }) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;ToolbarGroup&gt;</span>
    &lt;ToolbarTitle style={{ color: '#EEEEEE' }} text="Exercies for today" /&gt;
    &lt;IconMenu
      value="0"
      iconButtonElement={
        &lt;IconButton&gt;
          &lt;ContentFilter color="white" /&gt;
        &lt;/IconButton&gt;
      }
    &gt;
      {!_.isEmpty(exercises)
        ? _.map(exercises, <span class="fstat-no" title="function not covered" >ex</span>ec =&gt;
<span class="cstat-no" title="statement not covered" >            &lt;MenuItem</span>
              key={exercises.indexOf(exec)}
              value={exercises.indexOf(exec)}
              primaryText={exec.toUpperCase()}
            /&gt;
          )
        : &lt;MenuItem key={0} value={0} primaryText="NO EXERCISE FOR TODAY" /&gt;}
    &lt;/IconMenu&gt;
  &lt;/ToolbarGroup&gt;;
&nbsp;
Exercises.propTypes = {
  exercises: PropTypes.arrayOf(PropTypes.string)
};
&nbsp;
export default Exercises;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
