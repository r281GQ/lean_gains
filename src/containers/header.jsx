import React from 'react';
import Immutable from 'immutable';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { FlatButton, IconButton, FontIcon } from 'material-ui';
import {
  Toolbar,
  ToolbarGroup,
  ToolbarSeparator,
  ToolbarTitle,
  DropDownMenu,
  MenuItem,
  IconMenu,
  CircularProgress
} from 'material-ui';

import ActionHome from 'material-ui/svg-icons/action/home';
import Done from 'material-ui/svg-icons/action/done';
import ArrowBack from 'material-ui/svg-icons/navigation/arrow-back';
import ContentFilter from 'material-ui/svg-icons/content/filter-list';

import { goBack } from 'react-router-redux';
import currentWeightSelector from './../store/selectors/current_weight';
import isTrainingDay from './../store/selectors/exercises';
import { openSideBar } from './../store/actionCreators/app_action_creators';
import todayMacros from './../store/selectors/current_macros';

//TODO just saving indicator
const HeaderContainer = ({
  currentWeight,
  exercises,
  todaysMacros,
  openSideBar,
  isLoading,
  userName,
  goBack
}) => {
  return (
    <Toolbar>
      <ToolbarGroup firstChild={true}>
        <IconButton tooltip="Main menu" onTouchTap={() => openSideBar()}>
          <ActionHome />
        </IconButton>
      </ToolbarGroup>
      <ToolbarGroup>
        <IconButton onTouchTap={goBack} tooltip={`Go back`}>
          <ArrowBack />
        </IconButton>
      </ToolbarGroup>
      <ToolbarGroup>
        <ToolbarTitle text={`Welcome ${userName} `} />
      </ToolbarGroup>
      <ToolbarGroup>
        <ToolbarTitle text={`CurrentWeight ${currentWeight}`} />
        <ToolbarTitle text={`Macros: ${todaysMacros.calorie}`} />
        <ToolbarTitle
          text={`P: ${todaysMacros.protein} C: ${todaysMacros.carbohydrate} F: ${todaysMacros.fat}`}
        />
      </ToolbarGroup>

      <ToolbarGroup>
        {isLoading ? <CircularProgress /> : <Done />}
      </ToolbarGroup>
      <ToolbarGroup>
        <ToolbarTitle text="Exercies for today" />
        <IconMenu
          value={`0`}
          onChange={() => console.log()}
          iconButtonElement={
            <IconButton>
              <ContentFilter />
            </IconButton>
          }
        >
          {_.map(exercises, exec =>
            <MenuItem
              key={exercises.indexOf(exec)}
              value={exercises.indexOf(exec)}
              primaryText={exec}
            />
          )}
        </IconMenu>
      </ToolbarGroup>
    </Toolbar>
  );
};

const mapStateToProps = state => ({
  isLoading: state.getIn(['app', 'isLoading']),
  currentWeight: currentWeightSelector(state),
  exercises: isTrainingDay('main')(state).toJS(),
  todaysMacros: todayMacros(state).toJS(),
  userName: state.getIn(['userDetails', 'userName'])
});

const mapDispatchToProps = dispatch => ({
  goBack: () => dispatch(goBack()),
  openSideBar: () => dispatch(openSideBar())
});

export default connect(mapStateToProps, mapDispatchToProps)(HeaderContainer);
