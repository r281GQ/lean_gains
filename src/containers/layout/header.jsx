import React from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { goBack, goForward } from 'react-router-redux';
import Helmet from 'react-helmet';

import { openSideBar } from './../../store/actionCreators/app_action_creators';
import isTrainingDay from './../../store/selectors/exercises';
import todayMacros from './../../store/selectors/current_macros';
import UnauthanticatedHeader from './../../components/layout/unauth_header';
import AuthanticatedHeader from './../../components/layout/authenticated_header';
// import withDataCheck from './../enhancers/with_data_check';



const HeaderContainer = ({
  location: { pathname },
  exercises,
  todaysMacros,
  openSideBar,
  isLoading,
  userName,
  photo,
  isAuthenticated,
  goBack,
  goForward
}) =>
  <div>
    <Helmet>
      <meta charSet="utf-8" />
      <title>LeanGains</title>
      <link
        rel="icon"
        href="https://d30y9cdsu7xlg0.cloudfront.net/png/61493-200.png"
      />
      <link rel="canonical" href="http://mysite.com/example" />
    </Helmet>
    {!isAuthenticated
      ? <UnauthanticatedHeader pathname={pathname} />
      : <AuthanticatedHeader
          openSideBar={() => openSideBar()}
          todaysMacros={todaysMacros.toJS()}
          exercises={exercises.toJS()}
          isLoading={isLoading}
          userName={userName}
          photo={photo}
          goBack={goBack}
          goForward={goForward}
        />}
  </div>;

HeaderContainer.propTypes = {
  exercises: ImmutablePropTypes.list,
  todaysMacros: ImmutablePropTypes.map,
  openSideBar: PropTypes.func,
  isLoading: PropTypes.bool,
  userName: PropTypes.string,
  photo: PropTypes.string,
  isAuthenticated: PropTypes.bool
};

const mapStateToProps = state => ({
  isAuthenticated: state.getIn(['auth', 'authenticated']),
  isLoading: state.getIn(['app', 'isLoading']),
  exercises: isTrainingDay('main')(state),
  todaysMacros: todayMacros(state),
  userName: state.getIn(['userDetails', 'userName']),
  photo: state.getIn(['userDetails', 'picture'])
});

export default connect(mapStateToProps, { openSideBar, goBack, goForward })(
  withRouter(HeaderContainer)
);
