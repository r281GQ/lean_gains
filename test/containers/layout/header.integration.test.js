import React from 'react';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { mount } from 'enzyme';
import { PureHeaderContainer } from './../../../src/containers/layout/header';

const muiTheme = getMuiTheme();

const router = {
  history: {
    push: () => undefined,
    replace: () => undefined,
    createHref: () => undefined
  }
};

injectTapEventPlugin();

describe('Header integration test', () => {
  it('should render', () => {
    const component = mount(
      <PureHeaderContainer location={{ pathname: '' }} />,
      {
        context: { muiTheme, router },
        childContextTypes: {
          muiTheme: PropTypes.object,
          router: PropTypes.any
        }
      }
    );

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
