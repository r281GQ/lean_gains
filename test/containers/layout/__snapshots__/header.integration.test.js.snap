// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header integration test should render 1`] = `
<HeaderContainer
  location={
    Object {
      "pathname": "",
    }
  }
>
  <div>
    <HelmetWrapper
      encodeSpecialCharacters={true}
    >
      <SideEffect(NullComponent)
        encodeSpecialCharacters={true}
        link={
          Array [
            Object {
              "href": "[object Object]",
              "rel": "icon",
            },
          ]
        }
        meta={
          Array [
            Object {
              "charset": "utf-8",
            },
          ]
        }
        title="LeanGains"
        titleAttributes={Object {}}
      >
        <NullComponent
          encodeSpecialCharacters={true}
          link={
            Array [
              Object {
                "href": "[object Object]",
                "rel": "icon",
              },
            ]
          }
          meta={
            Array [
              Object {
                "charset": "utf-8",
              },
            ]
          }
          title="LeanGains"
          titleAttributes={Object {}}
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
    <UnauthanticatedHeader
      pathname=""
    >
      <Toolbar
        noGutter={false}
        style={
          Object {
            "backgroundColor": "#757575",
            "color": "#EEEEEE",
          }
        }
      >
        <div
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#757575",
              "boxSizing": "border-box",
              "color": "#EEEEEE",
              "display": "flex",
              "height": 56,
              "justifyContent": "space-between",
              "muiPrepared": true,
              "padding": "0px 24px",
            }
          }
        >
          <ToolbarGroup
            firstChild={false}
            lastChild={false}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "space-between",
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "muiPrepared": true,
                  "position": "relative",
                }
              }
            />
          </ToolbarGroup>
          <ToolbarGroup
            firstChild={false}
            lastChild={false}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "space-between",
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "muiPrepared": true,
                  "position": "relative",
                }
              }
            >
              <Link
                replace={false}
                to="/login"
              >
                <a
                  onClick={[Function]}
                >
                  <RaisedButton
                    disabled={false}
                    fullWidth={false}
                    label="Log in"
                    labelPosition="after"
                    primary={false}
                    secondary={false}
                  >
                    <Paper
                      circle={false}
                      rounded={true}
                      style={
                        Object {
                          "display": "inline-block",
                          "minWidth": 88,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }
                      transitionEnabled={true}
                      zDepth={1}
                    >
                      <div
                        style={
                          Object {
                            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                            "backgroundColor": "#ffffff",
                            "borderRadius": 2,
                            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                                   0 1px 4px rgba(0, 0, 0, 0.12)",
                            "boxSizing": "border-box",
                            "color": "rgba(0, 0, 0, 0.87)",
                            "display": "inline-block",
                            "fontFamily": "Roboto, sans-serif",
                            "minWidth": 88,
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          }
                        }
                      >
                        <EnhancedButton
                          containerElement="button"
                          disabled={false}
                          focusRippleColor="rgba(0, 0, 0, 0.87)"
                          focusRippleOpacity={0.1}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onKeyboardFocus={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          onTouchTap={[Function]}
                          style={
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderRadius": 2,
                              "height": 36,
                              "lineHeight": "36px",
                              "padding": 0,
                              "textAlign": "center",
                              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              "width": "100%",
                            }
                          }
                          tabIndex={0}
                          touchRippleColor="rgba(0, 0, 0, 0.87)"
                          touchRippleOpacity={0.1}
                          type="button"
                        >
                          <button
                            disabled={false}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            onTouchTap={[Function]}
                            style={
                              Object {
                                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                "backgroundColor": "#ffffff",
                                "border": 10,
                                "borderRadius": 2,
                                "boxSizing": "border-box",
                                "cursor": "pointer",
                                "display": "inline-block",
                                "fontFamily": "Roboto, sans-serif",
                                "fontSize": "inherit",
                                "fontWeight": "inherit",
                                "height": 36,
                                "lineHeight": "36px",
                                "margin": 0,
                                "muiPrepared": true,
                                "outline": "none",
                                "padding": 0,
                                "position": "relative",
                                "textAlign": "center",
                                "textDecoration": "none",
                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                "verticalAlign": null,
                                "width": "100%",
                                "zIndex": 1,
                              }
                            }
                            tabIndex={0}
                            type="button"
                          >
                            <TouchRipple
                              abortOnScroll={true}
                              color="rgba(0, 0, 0, 0.87)"
                              opacity={0.1}
                            >
                              <div
                                onMouseDown={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchStart={[Function]}
                              >
                                <div
                                  style={
                                    Object {
                                      "backgroundColor": false,
                                      "borderRadius": 2,
                                      "height": 36,
                                      "muiPrepared": true,
                                      "top": 0,
                                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                    }
                                  }
                                >
                                  <span
                                    style={
                                      Object {
                                        "color": "rgba(0, 0, 0, 0.87)",
                                        "fontSize": 14,
                                        "fontWeight": 500,
                                        "letterSpacing": 0,
                                        "margin": 0,
                                        "muiPrepared": true,
                                        "opacity": 1,
                                        "paddingLeft": 16,
                                        "paddingRight": 16,
                                        "position": "relative",
                                        "textTransform": "uppercase",
                                        "userSelect": "none",
                                      }
                                    }
                                  >
                                    Log in
                                  </span>
                                </div>
                              </div>
                            </TouchRipple>
                          </button>
                        </EnhancedButton>
                      </div>
                    </Paper>
                  </RaisedButton>
                </a>
              </Link>
              <ToolbarSeparator
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.4)",
                    "lineHeight": "56px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.175)",
                      "color": "rgba(0, 0, 0, 0.4)",
                      "display": "block",
                      "height": 32,
                      "lineHeight": "56px",
                      "marginLeft": 24,
                      "muiPrepared": true,
                      "width": 1,
                    }
                  }
                />
              </ToolbarSeparator>
              <GoogleLoginButton>
                <RaisedButton
                  disabled={false}
                  fullWidth={false}
                  href="/api/auth/google"
                  icon={
                    <FontIcon
                      className="fa fa-google"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    />
                  }
                  label="Log in with Google"
                  labelPosition="before"
                  primary={false}
                  secondary={false}
                >
                  <Paper
                    circle={false}
                    rounded={true}
                    style={
                      Object {
                        "display": "inline-block",
                        "minWidth": 88,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                    transitionEnabled={true}
                    zDepth={1}
                  >
                    <div
                      style={
                        Object {
                          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                          "backgroundColor": "#ffffff",
                          "borderRadius": 2,
                          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                                 0 1px 4px rgba(0, 0, 0, 0.12)",
                          "boxSizing": "border-box",
                          "color": "rgba(0, 0, 0, 0.87)",
                          "display": "inline-block",
                          "fontFamily": "Roboto, sans-serif",
                          "minWidth": 88,
                          "muiPrepared": true,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }
                    >
                      <EnhancedButton
                        containerElement="button"
                        disabled={false}
                        focusRippleColor="rgba(0, 0, 0, 0.87)"
                        focusRippleOpacity={0.1}
                        href="/api/auth/google"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        onTouchTap={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#ffffff",
                            "borderRadius": 2,
                            "height": 36,
                            "lineHeight": "36px",
                            "padding": 0,
                            "textAlign": "center",
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "width": "100%",
                          }
                        }
                        tabIndex={0}
                        touchRippleColor="rgba(0, 0, 0, 0.87)"
                        touchRippleOpacity={0.1}
                        type="button"
                      >
                        <a
                          disabled={false}
                          href="/api/auth/google"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          onTouchTap={[Function]}
                          style={
                            Object {
                              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                              "backgroundColor": "#ffffff",
                              "border": 10,
                              "borderRadius": 2,
                              "boxSizing": "border-box",
                              "cursor": "pointer",
                              "display": "inline-block",
                              "fontFamily": "Roboto, sans-serif",
                              "fontSize": "inherit",
                              "fontWeight": "inherit",
                              "height": 36,
                              "lineHeight": "36px",
                              "margin": 0,
                              "muiPrepared": true,
                              "outline": "none",
                              "padding": 0,
                              "position": "relative",
                              "textAlign": "center",
                              "textDecoration": "none",
                              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              "verticalAlign": "middle",
                              "width": "100%",
                              "zIndex": 1,
                            }
                          }
                          tabIndex={0}
                        >
                          <TouchRipple
                            abortOnScroll={true}
                            color="rgba(0, 0, 0, 0.87)"
                            opacity={0.1}
                          >
                            <div
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                            >
                              <div
                                style={
                                  Object {
                                    "backgroundColor": false,
                                    "borderRadius": 2,
                                    "height": 36,
                                    "muiPrepared": true,
                                    "top": 0,
                                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                  }
                                }
                              >
                                <span
                                  style={
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontSize": 14,
                                      "fontWeight": 500,
                                      "letterSpacing": 0,
                                      "margin": 0,
                                      "muiPrepared": true,
                                      "opacity": 1,
                                      "paddingLeft": 16,
                                      "paddingRight": 8,
                                      "position": "relative",
                                      "textTransform": "uppercase",
                                      "userSelect": "none",
                                    }
                                  }
                                >
                                  Log in with Google
                                </span>
                                <FontIcon
                                  className="fa fa-google"
                                  color="rgba(0, 0, 0, 0.87)"
                                  onMouseEnter={[Function]}
                                  onMouseLeave={[Function]}
                                  style={
                                    Object {
                                      "marginLeft": 0,
                                      "marginRight": 12,
                                      "verticalAlign": "middle",
                                    }
                                  }
                                >
                                  <span
                                    className="fa fa-google"
                                    color="rgba(0, 0, 0, 0.87)"
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    style={
                                      Object {
                                        "color": "rgba(0, 0, 0, 0.87)",
                                        "display": "inline-block",
                                        "fontSize": 24,
                                        "marginLeft": 0,
                                        "marginRight": 12,
                                        "muiPrepared": true,
                                        "position": "relative",
                                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                        "userSelect": "none",
                                        "verticalAlign": "middle",
                                      }
                                    }
                                  />
                                </FontIcon>
                              </div>
                            </div>
                          </TouchRipple>
                        </a>
                      </EnhancedButton>
                    </div>
                  </Paper>
                </RaisedButton>
              </GoogleLoginButton>
            </div>
          </ToolbarGroup>
        </div>
      </Toolbar>
    </UnauthanticatedHeader>
  </div>
</HeaderContainer>
`;
